Amazon.IonDotnet/Internals/Text/RawTextReader.cs

```diff
@@ -735,7 +735,31 @@ public int CurrentDepth
 
         public abstract int GetBytes(Span<byte> buffer);
 
-        public IEnumerable<SymbolToken> GetTypeAnnotations()
+        public string[] GetTypeAnnotations()
+        {
+            List<string> annotations = new List<string>();
+            foreach (SymbolToken symbolToken in _annotations)
+            {
+                if (symbolToken.Text is null && symbolToken.ImportLocation != default)
+                {
+                    ISymbolTable symtab = GetSymbolTable();
+                    if (symbolToken.ImportLocation.Sid < -1 || symbolToken.ImportLocation.Sid > symtab.MaxId)
```

Comment:
It's not equivalent. I did want to clarify, according to the docs: 

> Symbol Zero
SID zero (i.e. $0) is a special symbol that is not assigned text by any symbol table, even the system symbol table. Symbol zero always has unknown text, and can be useful in synthesizing symbol identifiers where the text image of the symbol is not known in a particular operating context.

> It is important to note that $0 is only semantically equivalent to itself and to locally-declared SIDs with unknown text. It is not semantically equivalent to SIDs with unknown text from shared symbol tables, so replacing such SIDs with $0 is a destructive operation to the semantics of the data.

In this case, `GetSymbolTable` returns a system table, so is it correct that symbol zero with a null text should go through? If I refactor this to a similar workflow where doing a lookup via `symtab.FindKnownSymbol(symbolToken.ImportLocation.Sid)` and do a null check, it would throw in cases where symbol zero is encountered. Doing so, would also cause prior unit tests to fail.

Could you clarify the expected behaviour?

Reply:
`GetTypeAnnotations()` should fail upon encountering any symbol with unknown text, including symbol zero. If the user intends to handle symbol zero, they must use `GetTypeAnnotationSymbols()`.

Reply:
Fixed.